src/create-resource.js

```diff
@@ -0,0 +1,89 @@
+import { useState, useEffect, useCallback, useContext } from "react";
+import uuid from "uuid/v1";
+import { curry } from "ramda";
+import store, { selectResource, RECEIVE_STATE } from "./store";
+import Context from "./Context";
+
+const createResource = curry(
+  (entryGetter, entrySetter, entryPredicate, loader) => {
+    const resourceId = uuid();
+
+    const getResourceState = () =>
+      selectResource(store.getState(), { resourceId });
+
+    const setResourceState = nextState => {
+      store.dispatch({
+        type: RECEIVE_STATE,
+        resourceId,
+        state:
+          typeof nextState === "function"
+            ? nextState(getResourceState())
+            : nextState
+      });
+    };
+
+    const setEntryState = curry((args, state) =>
+      setResourceState(entrySetter(getResourceState(), args, state))
+    );
+
+    const subscribe = onChange => {
+      let currentState = getResourceState();
+      return store.subscribe(() => {
+        const nextResourceState = getResourceState();
+        if (nextResourceState !== currentState) {
+          currentState = nextResourceState;
+          onChange();
+        }
+      });
+    };
+
+    const pendingLoaders = new Map();
+
+    return {
+      id: resourceId,
+      getState: getResourceState,
+      setState: setResourceState,
+      refresh: (...args) => loader(...args).then(setEntryState(args)),
```

Comment:
**Question**

What would happen if this fails, does the refresh throw a promise, or does the user need to handle it when they call it?

Reply:
Yeah, they would need to handle it. I think the only time you're going to do a `refresh` is when you have already fetched the data but you think it's out of date. If we were to have the `RemoteResourceBoundary` catch the error from a rejected promise and render the error UI that would take them away from viewing the data they already have. In this case I would imagine they would rather show some alert or a toast or something to tell the user the refresh failed, but not block them from using the data they already have.

Does that make sense?

Reply:
Absolutely! I'll make a note for the resource in the documentation that this is the case that it is not caught!

Reply:
Thanks!

Reply:
I went ahead and made a note in the README for this.